corehq/apps/export/models/new.py

```diff
@@ -0,0 +1,342 @@
+            next_doc = doc.get(path[0], {})
+            if type(next_doc) == list:
+                new_docs.extend(next_doc)
+            else:
+                new_docs.append(next_doc)
+        return self._get_items_for_repeat(path[1:], new_docs)
+
+
+class ExportInstance(Document):
+
+    class Meta:
+        app_label = 'export'
+
+
+class ExportRow(object):
+    def __init__(self, data):
+        self.data = data
+
+
+class ScalarItem(ExportItem):
+    """
+    A text, numeric, date, etc. question or case property
+    """
+
+
+class Option(DocumentSchema):
+    """
+    This object represents a multiple choice question option.
+
+    last_occurrence is an app build number representing the last version of the app in
+    which this option was present.
+    """
+    last_occurrence = IntegerProperty()
+    value = StringProperty()
+
+
+class MultipleChoiceItem(ExportItem):
+    """
+    A multiple choice question or case property
+    Choices is the union of choices for the question in each of the builds with
+    this question.
+    """
+    options = SchemaListProperty(Option)
+
+    @classmethod
+    def create_from_question(cls, question, appVersion):
+        item = super(MultipleChoiceItem, cls).create_from_question(question, appVersion)
+
+        for option in question.options:
+            item.options.append(Option(
+                last_occurrence=appVersion,
+                value=option['value']
+            ))
+        return item
+
+    @classmethod
+    def merge(cls, one, two):
+        item = super(MultipleChoiceItem, cls).merge(one, two)
+        options = _merge_lists(one.options, two.options,
+            keyfn=lambda i: i.value,
+            resolvefn=lambda option1, option2:
+                Option(
+                    value=option1.value,
+                    last_occurrence=max(option1.last_occurrence, option2.last_occurrence)
+                ),
+            copyfn=lambda option: Option(option.to_json())
+        )
+
+        item.options = options
+        return item
+
+
+class ExportGroupSchema(DocumentSchema):
+    """
+    An object that represents a logical group of questions in the form
+    """
+    path = ListProperty()
+    items = SchemaListProperty(ExportItem)
+    last_occurrence = IntegerProperty()
+
+
+class ExportDataSchema(DocumentSchema):
+    """
+    An object representing the things that can be exported for a particular
+    form xmlns or case type.
+    Each item in the list is uniquely identified by its path and doc_type.
+    repeats is a list of RepeatGroups, present at any level of the question hierarchy
+    """
+    group_schemas = SchemaListProperty(ExportGroupSchema)
+    datatype_mapping = defaultdict(lambda: ScalarItem, {
+        'MSelect': MultipleChoiceItem,
+    })
+
+    @staticmethod
+    def generate_schema_from_builds(domain, app_id, unique_form_id):
+        app_build_ids = get_built_app_ids_for_app_id(domain, app_id)
+        all_xform_conf = TableConfiguration()
```

Comment:
yep, currently no test tests this function (which is why this is wrong), going to work on adding some tests to it